packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/nodeaction/ManageVisibilityNodeActionProvider.java

```diff
@@ -0,0 +1,44 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.sirius.web.application.nodeaction;
+
+import org.eclipse.sirius.components.collaborative.diagrams.api.IActionsProvider;
+import org.eclipse.sirius.components.collaborative.diagrams.dto.Action;
+import org.eclipse.sirius.components.core.api.IEditingContext;
+import org.eclipse.sirius.components.diagrams.IDiagramElement;
+import org.eclipse.sirius.components.diagrams.Node;
+import org.eclipse.sirius.components.diagrams.description.DiagramDescription;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+/**
+ * Used to provide the show label node action.
+ *
+ * @author mcharfadi
+ */
+@Service
+public class ManageVisibilityNodeActionProvider implements IActionsProvider {
```

Comment:
I somehow missed that it was activated this aggressively on every node with child nodes. We just can't hardcode that decision, we will let specifiers choose where they want to provide it. I will thus restrict this to the flow topography diagrams